[RPMSG] Adding Python transport layer for RPMsg sysfs kernel module

MarekNovakNXP opened the pull request:
- Adds support for RPMsg sysfs interface
- Python driver for RPMsg sysfs interface added in a zip file, to be uploaded to pypi or github...
- RPMsg sysfs kernel module patch added, applicable to version 4.1 of Freescale community Yocto

[rpmsg.zip](https://github.com/EmbeddedRPC/erpc/files/477113/rpmsg.zip)
[rpmsg_sysfs_interface.zip](https://github.com/EmbeddedRPC/erpc/files/477116/rpmsg_sysfs_interface.zip)

flit commented on erpc_python/erpc/transport.py:
```diff
@@ -29,6 +29,8 @@
 
 import struct
 import serial
+from rpmsg.sysfs import RpmsgEndpoint
```
This import will fail if the rpmsg package is not available. Please move it below the rest of the imports and wrap in a try-except statement. You can set a boolean variable to track whether `RpmsgEndpoint` is available, and use it in `RpmsgTransport`.

flit commented on erpc_python/erpc/transport.py:
```diff
@@ -105,6 +107,26 @@ def _base_receive(self, count):
 class ConnectionClosed(Exception):
     pass
 
+class RpmsgTransport(Transport):
```
Make `RpmsgTransport` a subclass of `FramedTransport`. This will allow you to avoid having to do an arbitrary 2048-byte receive. Instead, the `FramedTransport` will do fixed length receives of the 4-byte frame header followed by the known message length.

MarekNovakNXP commented:
I have modified this pull request as the RPMsg Python wrapper has changed.
There is no need to specify an arbitrary length now and so there is no need for it being a subclass of FramedTransport. FramedTransport would add some overhead, which is not needed in case of RPMsg.

Added try-except for import of RPMsg.

[0001-RPMSG-Adding-Python-transport-layer-for-RPMsg-sysfs-.patch.zip](https://github.com/EmbeddedRPC/erpc/files/541152/0001-RPMSG-Adding-Python-transport-layer-for-RPMsg-sysfs-.patch.zip)

hakehuang commented:
can we possible make it installed by pip install?

flit commented:
@hakehuang Yes, thanks for reminding me. I forgot to upload to pypi.

Hadatko commented:
Please provide patch aligned with latest eRPC code on develop branch.

MarekNovakNXP commented:
Closed, replaced by:
[RPMSG] [Python] Adding RPMsg Multiendpoint kernel module support #4
